tests/common/fixtures/elasticsearch.py

```diff
@@ -15,14 +16,20 @@
 def es_client():
     client = _es_client()
     yield client
+    # Propogate changes to all shards to make sure all annotations that were added get removed.
```

Comment:
It's not clear to me but I think there may be a misunderstanding of what the refresh operation does. I was able to reproduce the issue even when Elasticsearch was configured to use only a single shard for the index. My understanding is that refreshing is an operation that happens on each shard separately and makes sure that any changes written to that shard are visible when that shard is later queried.

- Details on the relation between indexes, shards and Lucene indexes: https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started-concepts.html
- Some details on how Near Real Time search + reopening works on a single Lucene index: http://blog.mikemccandless.com/2011/06/lucenes-near-real-time-search-is-fast.html

Reply:
I've re-read the docs and I think it has to do with the translog not pushing the change to the segment. When a refresh happens it pushes it to the segment but otherwise it's kept in the change log and it only becomes available for search when it's in the segment.